### PR TITLE
Feature/joomla

### PR DESCRIPTION
**Pull request by SolomonShorser-OICR**
FINALLY wrapped up most of the code to get the reactome-docker project compatible with the new Joomla CMS.

This PR isn't looking for detailed code review (although code review is OK) so much as it is a request for testing that this works on someone else's system.

### CODE REVIEW

**Comment by SolomonShorser-OICR** [comment]
@jweiser I realise this is a very large PR that spans over 2 years of development. I think it would be easier if this code could just get merged back in to develop and then new, smaller branches could be used for future development efforts, but now that reviews are required, someone needs to click the "approve" button before this can go ahead.

**Comment by jweiser** [comment]
> @jweiser I realise this is a very large PR that spans over 2 years of development. I think it would be easier if this code could just get merged back in to develop and then new, smaller branches could be used for future development efforts, but now that reviews are required, someone needs to click the "approve" button before this can go ahead.

Has the codebase in this branch been reviewed at all, though?  I'm a little uncomfortable merging so much code into the develop branch if it hasn't been looked at.

You also mention in the description of the PR that its purpose is to test if it works on someone else's system.  Has it been tested by someone else?

**Comment by SolomonShorser-OICR** [comment]
@jweiser Some of this was written before we implemented standard code reviews. Originally, testing on someone else's system was a goal, but I don't think there's as much interest in that as there used to be (unless you're willing to try it!). The three separate stand-alone images seems to be the way most users will use Reactome-in-docker - at least in the short-term. BUT... everything is still on this one branch which is rather silly. I feel it should just be merged and then smaller branches for more focused development can be used.

You do not _need_ to review this in detail (or at all, really), but PRs for review opportunity was requested, and it is still here. I could use my Admin privileges and _force_ a merge, but I didn't think that would be very nice.

**Comment by jweiser** [comment]
> @jweiser Some of this was written before we implemented standard code reviews. Originally, testing on someone else's system was a goal, but I don't think there's as much interest in that as there used to be (unless you're willing to try it!). The three separate stand-alone images seems to be the way most users will use Reactome-in-docker - at least in the short-term. BUT... everything is still on this one branch which is rather silly. I feel it should just be merged and then smaller branches for more focused development can be used.
> 
> You do not _need_ to review this in detail (or at all, really), but PRs for review opportunity was requested, and it is still here.
> 
> I _could_ use my Admin privileges and _force_ a merge, but I didn't think that would be very nice. ;)

I could try running it.  What is the minimum memory and CPU needed?

**Comment by SolomonShorser-OICR** [comment]
Not sure about memory, I'd estimate at least 10-15 GB to build, probably the same to run.

**Comment by SolomonShorser-OICR** [comment]
If you're going to try to build the whole thing, I would set it up to start as the last thing you do for the work day, and let it run overnight. ;) It will take a while, and it might not even be _able_ to run to completion on our laptops.

But at this point, I really think the focus should be on the three stand-alone docker images.

**Comment by jweiser** [comment]
> Not sure about memory, I'd estimate at least 10-15 GB to build, probably the same to run.

I only have 11.5GB of RAM on my machine total, so I don't think I'd be able to run it.  Maybe you ask Justin or Robin if they could give it a try.

I'd feel more comfortable about merging this amount of code after it was tried on another system by a user who isn't familiar with the code..

**Comment by SolomonShorser-OICR** [comment]
@jweiser I think Justin's built the images. I've asked him to comment here (Can't mention him with `@cookersjs` notation, for some reason).

**Comment by jweiser** [comment]
> @jweiser I think Justin's built the images. I've asked him to comment here (Can't mention him with `@cookersjs` notation, for some reason).

The access to this repo was only certain people.  I added the reactome "developers" list, so all of us should now be able to be tagged for mentions, reviews, etc.

**Comment by cookersjs** [comment]
Yes I've been able to build it. If I recall correctly, in the early going it had a few speed bumps that needed to be addressed but it was able to get to a point where it could be built from scratch. I would have done it on this laptop, which only has 8GBs of RAM. @SolomonShorser-OICR given that it's been a couple of months would you like me to try and building it agin to confirm that it works? 

I think I understand the desire to have it merged after all this time and I imagine that with the amount of code here there will inevitably be discussions that delay the merging even longer. Would a fair compromise be to perform the merge but keep the branch alive for now? This code is fairly isolated within the repository isn't it? Are their other tasks/code that this repository performs that this merge's new code alters?

**Comment by SolomonShorser-OICR** [comment]
@jweiser Ah, thanks for that!

**Comment by SolomonShorser-OICR** [comment]
The code on dev/master is for the old WordPress version of Reactome, which was turned out to be an epic-scale task. All future work will need to be based on the work already on this branch. This PR was first opened about a year ago, when I finally got some consistently good builds using the new Joomla version of Reactome, but it never got merged and everything after that has continued to pile into this branch.

What would be the point of keeping the branch active after merging? The main goal of this branch was to switch to Joomla, and that has been accomplished.

**Comment by cookersjs** [comment]
I was thinking for retroactive review, if we wanted to revisit the changes that this branch brought. That being said, I guess we can just look over the closed PR. And if this branch has become the de facto master, than I agree that having changes continue to go here is just going to become more and more unwieldy

**Comment by SolomonShorser-OICR** [comment]
So? Thoughts? Can we merge this yet?

**Comment by cookersjs** [comment]
@SolomonShorser-OICR Crap, my bad. I've been working on other things and this completely slipped through the cracks. I'm going to run it today and I'll give you a firm answer after the weekend.

**Comment by cookersjs** [comment]
So it turned out that building the AnalysisCore with only 6GB RAM allocated to Docker was not enough, but the build itself didn't have any problems. I still think we're good to merge -- once we are back in the office we can run it on our workstations from scratch

**Comment by SolomonShorser-OICR** [comment]
@jweiser You've got a more powerful laptop, have you tried building _any_ of these?

**Comment by jweiser** [comment]
> @jweiser You've got a more powerful laptop, have you tried building _any_ of these?

@SolomonShorser-OICR I haven't.  I've just been swamped with other tasks.  I don't think my laptop will be able to run it either if more than 6GB of RAM is required.  I only have my older laptop at home.

File: analysis-core/entrypoint.sh
Comment (cookersjs) [inline analysis-core/entrypoint.sh:14]: One consideration would be to remove the '-v' flag from running this. Considering the amount of logging output being generated when we are building the docker images (if that is all this is used for), I think it hurts more than it helps because they are almost entirely just update logs, and there is a lot of updates!

Comment (SolomonShorser-OICR) [reply analysis-core/entrypoint.sh:14]: Maybe there needs to be a happy medium between SO MUCH OUTPUT and nothing. I think I added the `-v` so that if there was any error, there would be some context to it. But if there's too much output, we can take that out.

Comment (cookersjs) [reply analysis-core/entrypoint.sh:14]: Yea, your right. The `verbose` switch is the only way to get detailed logging of the analysis-core -- the default logs produced are only for STDERR. It would be nice if the stdout logs were saved as well, and then the jenkins view would only have the 'progress' (and errors) of the docker build, but not the INFO. Ah well 🤷‍♂️

**Review by cookersjs — COMMENTED** [review]
Just quickly looked over the changes since I last reviewed - only comment would be that some `RELEASE_VERSION` variables are set to Release74 and others are set to Release75. 

As an idea - what about setting the default value to `ReleaseXX`. That way if an image gets built that has that XX tag you would at least know that the dockerfiles didn't get that value when being built. Otherwise, just changing them all to have the same version number would be good.

**Review by cookersjs — DISMISSED** [review]
I approve in case you wanted to merge. Its a HUGE merge though, so might worth discussing as a group

**Comment by SolomonShorser-OICR** [comment]
> I approve in case you wanted to merge. Its a HUGE merge though, so might worth discussing as a group

Yes, it is a big merge, but the branch and PR were created before our current review standards, so at the time, I didn't expect anyone else to review other than myself - as we were all doing back then.

**Review by gsviteri — DISMISSED** [review]

**Pull request merged**
